src/semidiscretization.jl

```diff
@@ -159,6 +159,27 @@ function integrate_quantity!(quantity,
     integrate_quantity(func, q, semi)
 end
 
+# The entropy/energy of the Serre-Green-Naghdi equations
+# takes the whole `q` for every point in space since it requires
+# the derivative of the velocity `v_x`.
+function integrate_quantity(func::Union{typeof(energy_total),
```

Comment:
Again, if we rename these functions to `*_modified` we would not need the additional methods since they are already covered by the above method.